File: algorithms/cpp/Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ret = 0;
        int n = prices.size();
        if(n == 0 || n == 1) return ret;
        
        int minp = prices[0];
        for(int i = 1;i < n;++i) {
            if(prices[i] > minp) {
                ret += prices[i] - minp;
            } 
            minp = prices[i];
        }
        return ret;
    }
};

File: algorithms/java/Best Time to Buy and Sell Stock II.java
public class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 0) return 0;
        int profit = 0;
        int cur = prices[0];
        for(int i = 1;i < prices.length;++i) {
            if(prices[i] > cur) {
                profit += prices[i] - cur;
                cur = prices[i];
            } else {
                cur = prices[i];
            }
        }
        
        return profit;
    }
}